proto/redpanda/api/dataplane/v1alpha1/kafka_connect.proto

```diff
@@ -0,0 +1,288 @@
+    (buf.validate.field).required = true
+  ];
+}
+
+message ResumeConnectorRequest {
+  string cluster_name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+  string name = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+}
+
+message GetConnectorRequest {
+  string cluster_name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+  string name = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+}
+
+message CreateConnectorRequest {
+  string cluster_name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+  ConnectorSpec connector = 2;
+}
+
+message GetConnectorResponse {
+  ConnectorSpec connector = 1;
+  ConnectorState state = 4;
+  repeated ConnectorError errors = 5;
+}
+
+message CreateConnectorResponse {
+  string name = 1;
+  // config is the connector json configuration map, every key is a string
+  map<string, string> config = 2;
+  repeated TaskInfo tasks = 3 [(google.api.field_behavior) = OUTPUT_ONLY];
+  string type = 4 [(google.api.field_behavior) = OUTPUT_ONLY];
+}
+
+message ListConnectorsResponse {
+  repeated ConnectorInfoStatus connectors = 1;
+
+  message ConnectorInfoStatus {
+    string name = 1;
+    optional ConnectorSpec info = 2;
+    optional ConnectorStatus status = 3;
+    //holistic_state of all the tasks within the connector this is our internal
+    //holistic state concept
+    ConnectorState holistic_state = 4;
+    //Errors is list of parsed coonectors' and taks' errors
+    repeated ConnectorError errors = 5;
+  }
+}
+
+// KafkaConnectService is the service for the kafka-connect, it exposes the
+// kafka-connect api, you can set mutiple kafka-connect services and all of
+// them can be managed using this service definition, the request is not only
+// proxied but also enriched with better error handling and custom
+// documentation and configuration
+service KafkaConnectService {
+  //ListConnectors implements a custom version of the kafka-connect api,
+  //instead of a map it returns a list
+  rpc ListConnectors(ListConnectorsRequest) returns (ListConnectorsResponse) {
+    option (google.api.http) = {
+      get: "/v1alpha1/{cluster_name}/connectors"
+      response_body: "connectors"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "List connectors"
+      description: "List connectors managed by the kafka-connect service, use the cluster_name defined in the configuration file."
+    };
+  }
+  //CreateConnector implements a the create connector method, and exposes an
+  //equivalent REST endpoint as the kafka connect API endpoint
+  rpc CreateConnector(CreateConnectorRequest) returns (CreateConnectorResponse) {
+    option (google.api.http) = {
+      post: "/v1alpha1/{cluster_name}/connectors"
```

Comment:
Here I am not prefixing the endpoints with anything, should we use `kafka-connect` or leave like this and use the kafka connect cluster name/s set in the config file as prefix? cc @weeco

Reply:
Yes, let's use `kafka-connect` as a prefix. Even though we want to use managed connectors as a name this API has an impact beyond cloud (self-hosted, community) and I think it's fair to say that this is a kafka connect API, not necessarily a managed connectors API. 

@vladoschreiner